.github/workflows/codeql.yml

```diff
@@ -5,22 +5,40 @@ on:
     branches: ['main', '*/main']
 
   pull_request:
+    types: [opened, synchronize, labeled]
 
 jobs:
   analyze:
     name: Analyze
     runs-on: ubuntu-latest
+    # We run codeql only if it's:
+    #   1) pull request not from a fork (ie. internal PR), or
+    #   2) pull request from a fork (ie. external PR) that was added "run-codeql" label
+    if: |
+      (github.event.pull_request.head.repo.full_name == github.repository) ||
+      (github.event.action == 'labeled' && github.event.label.name == 'run-codeql')
```

Comment:
```suggestion
```

Actually, these can be removed, unlike our normal `run-tests` label logic, because this codeql doesn't use any repo secrets and we can be more relaxed here.